terragrunt/modules/sync-team/confirmation.tf

```diff
@@ -0,0 +1,132 @@
+locals {
+  domain = "confirmation.sync-team-prod.rust-lang.net"
+}
+
+data "aws_ssm_parameter" "confirmation_parameters" {
+  for_each = toset([
+    "/prod/sync-team-confirmation/github-oauth-client-id",
+    "/prod/sync-team-confirmation/github-oauth-client-secret",
+  ])
+
+  name = each.value
+  // We don't need the actual value, just their ARNs:
+  with_decryption = false
+}
+
+module "lambda_confirmation" {
+  source = "terraform-aws-modules/lambda/aws"
+
+  function_name = "sync-team-confirmation"
+  description   = "Out-of-band confirmation before applying sync-team changes."
+  handler       = "index.handler"
+  runtime       = "python3.9"
+  publish       = true
+
+  source_path = "lambdas/sync-team-confirmation"
+
+  create_lambda_function_url = true
+
+  // While the module supports attaching an inline policy here, in practice we
+  // cannot attach it here as it'd create a circular dependency between the
+  // CodeBuild build and the Lambda policy.
+}
+
+resource "aws_iam_policy" "lambda_confirmation" {
+  name        = "sync-team-confirmation"
+  description = "Policy for the sync-team-confirmation Lambda"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect   = "Allow"
+        Action   = "codebuild:StartBuild"
+        Resource = aws_codebuild_project.sync_team.arn
+      },
+      {
+        Effect   = "Allow"
+        Action   = ["ssm:GetParameter", "ssm:GetParameters"],
+        Resource = [for p in data.aws_ssm_parameter.confirmation_parameters : p.arn]
+      },
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_confirmation" {
+  role       = module.lambda_confirmation.lambda_role_name
+  policy_arn = aws_iam_policy.lambda_confirmation.arn
+}
+
+module "confirmation_certificate" {
+  source  = "../acm-certificate"
+  providers = {
+    aws = aws.us-east-1
+  }
+  domains = [local.domain]
+  legacy  = false
+}
+
+resource "aws_cloudfront_distribution" "confirmation" {
```

Comment:
I'm surprised by wanting an entire cloudfront distribution. Is the reason for this getting the http to https redirect?

Reply:
The cloudfront distribution is needed to have the custom domain, which is less phishable than the random lambda URL.

Reply:
I don't think a CF distribution is a problem, seems pretty lightweight from most perspectives. I think this is fine for now.